Server/ConcordiaCurriculumManager/Services/CourseService.cs

```diff
@@ -367,4 +368,20 @@ private async Task VerifyCourseIsNotInCourseGroupingOrThrow(Course course)
                 $"A deletion request cannot be made for {course.Subject} {course.Catalog} as it is part of the {grouping.Name} course grouping"
             );
     }
+
+    public async Task<Course> GetCourseByIdAsync(Guid id)
+    {
+        var course = await _courseRepository.GetCourseByIdAsync(id);
+        if (course == null)
+        {
+            throw new NotFoundException($"Course with ID: {id} was not found.");
+        }
+
+        if (course.CourseState == CourseStateEnum.Deleted)

```

Comment:
Can you take out this check? The purpose of this endpoint is primarily so we can view course data from the dossier review page, so we should be able to see courses even that have been deleted (because some of the courses linked in the dossier review page will be part of deletion requests.)